flocker/node/script.py

```diff
@@ -50,14 +52,24 @@ class ZFSAgentOptions(Options):
          "The port on the control service to connect to.", int],
     ]
 
-    def parseArgs(self, hostname, host):
-        # Passing in the 'hostname' (really node identity) via command
-        # line is a hack.  See
-        # https://clusterhq.atlassian.net/browse/FLOC-1381 for solution.
-        self["hostname"] = unicode(hostname, "ascii")
+    def parseArgs(self, host):
         self["destination-host"] = unicode(host, "ascii")
 
 
+def _get_external_ip(host, port):
+    sock = socket()
+    sock.connect_ex((host, port))
+    return sock.getaddr()[1]  # XXX approximately
```

Comment:
Hopefully, yeah. At the moment calculation logic is still using `hostname` though so we still need to get it somehow.